static/script/devices/mediaplayer/samsung_streaming.js

```diff
@@ -415,6 +418,11 @@ require.def(
                 if (this.getState() !== MediaPlayer.STATE.BUFFERING) {
                     return;
                 }
+                
+                if (!this._isInitialBufferingFinished() && this._nextSeekingTo !== null) {
+                    this._deferSeekingTo = this._nextSeekingTo;
```

Comment:
As mentioned already this is all a workaround for problem with different events handling in SEF compared to Device API.
_deferSeekingTo is set to null for next seeking (playFrom during BUFFERING) in wrong time compared to what is expected by the app. In unit tests it is OK, but platform sometimes fires next event too early/late.
As a result in method _onFinishedBuffering() _deferSeekingTo still has some value and app stays in BUFFERING, but when app execution gets to _deferredSeek(), it is already null and app does not switch to PLAYING anymore.